central/auth/datastore/datastore_impl.go

```diff
@@ -141,12 +164,30 @@ func (d *datastoreImpl) InitializeTokenExchangers() error {
 			tokenExchangerErrors = errors.Join(tokenExchangerErrors, err)
 			continue
 		}
-		if err := d.set.UpsertTokenExchanger(exchanger, config.GetId()); err != nil {
-			tokenExchangerErrors = errors.Join(tokenExchangerErrors, err)
-		}
+		d.set.UpsertTokenExchanger(exchanger, config.GetId())
 	}
 	if tokenExchangerErrors != nil {
 		return tokenExchangerErrors
 	}
 	return tokenExchangerErrors
 }
+
+// wrapRollback wraps the error with potential rollback errors.
+// In the case the giving config is not nil, it will attempt to rollback the exchanger in the set in addition to
+// rolling back the transaction.
+func (d *datastoreImpl) wrapRollback(ctx context.Context, tx *pgPkg.Tx, err error, existingConfig *storage.AuthMachineToMachineConfig) error {
+	var exchangerErr error
+	if existingConfig != nil {
+		exchangerErr = d.set.RollbackExchanger(existingConfig.GetIssuer())
+	}
+
+	rollbackErr := tx.Rollback(ctx)
+	if exchangerErr != nil {
+		err = pkgErrors.Wrapf(exchangerErr, "rolling back due to exchanger error: %v", err)
```

Comment:
This message **reads** as if we are rolling back due to error coming from `d.set.RollbackExchanger`, but in reality we rollback due to `err`

Reply:
Hope this makes things more clear?